### PR TITLE
Remove thread hop to producer thread for trx signature recovery

### PR DESCRIPTION
**Pull request by heifner**
Remove the use of a `std::future` and `wait` in producer thread pool. Now after signature recovery is complete the chain thread posts directly to the main application thread. This improves the eosio token transfer per second (TPS) tests from ~31K TPS to ~36K TPS.

Transaction now transitions from net-thread => chain-thread => main-thread.

Removing the thread hop to chain-thread and performing key recovery on the net thread is possible (net-thread => main-thread), but testing shows it easily overwhelms the main-thread. Instead I think we need to create a memory pool of trxs that the main thread can pull from instead of pushing low priority tasks for trx execution to the main thread. That would greatly reduce the contention on the app post call and on the main thread priority queue. See https://github.com/AntelopeIO/leap/issues/1860

This builds off the improvement of #1846.

Removes `--producer-threads` option as there is now no use for the producer thread pool.

Issue #1690

### CODE REVIEW

File: plugins/producer_plugin/producer_plugin.cpp [hunk 1]
```diff
@@ -825,38 +815,57 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
...
+                                                           if (this->in_producing_mode()) {
+                                                              this->schedule_maybe_produce_block(true);
+                                                           } else {
+                                                              this->restart_speculative_block();
+                                                           }
+                                                        }
+                                                     }
+                                                     CATCH_AND_CALL(exception_handler);
+                                                  });
+                         } );
```
Comment (greg7mdp) [inline plugins/producer_plugin/producer_plugin.cpp:868]: I think it would be better to remove all occurences of this-> in this code.

**Review by greg7mdp — APPROVED** [review]

File: plugins/producer_plugin/producer_plugin.cpp (hunk 1 above)
Comment (heifner) [reply plugins/producer_plugin/producer_plugin.cpp]: Removed

File: plugins/producer_plugin/producer_plugin.cpp [hunk 2]
```diff
@@ -825,38 +815,57 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
...
-                                          } else {
-                                             self->restart_speculative_block();
-                                          }
-                                       }
-                                    }
-                                    CATCH_AND_CALL(exception_handler);
-                                 });
-                           }
-                        });
+      boost::asio::post( chain.get_thread_pool(),
```
Comment (linh2931) [inline plugins/producer_plugin/producer_plugin.cpp:818]: This section of code is deep nested and becomes complicated. Can more comments be added and/or it be refactored a bit?

Comment (heifner) [reply plugins/producer_plugin/producer_plugin.cpp]: Cleaned it up some.

**Review by greg7mdp — APPROVED** [review]

**Review by linh2931 — APPROVED** [review]

**Pull request merged**